File: Domain/Room.cs
namespace Domain
{
    public class Room
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Image { get; set; }
        public RoomType Type { get; set; } = RoomType.Public;
        public int Capacity { get; set; } = 50;
    }
}

File: client-app/src/app/models/room.ts
export interface Room {
  id: string;
  title: string;
  description: string;
  image: string;
  type: number;
  capacity: number;
}

export class Room implements Room{
  constructor(init?: RoomFormValues) {
    Object.assign(this, init);
  }
}

export class RoomFormValues {
  id?: string;
  title: string = '';
  description: string = '';
  image: string = '';
  type?: number = 0;
  capacity: number = 10;


  /**
   *
   */
  constructor(room?: RoomFormValues) {
    if(room){
      this.id = room.id;
      this.title = room.title;
      this.description = room.description;
      this.capacity = room.capacity;
      this.type = room.type;
    }
  }
}